heudiconv/bids.py

```diff
@@ -291,25 +292,27 @@ def get_formatted_scans_key_row(item):
         [ISO acquisition time, performing physician name, random string]
 
     """
-    dcm_fn = item[-1][0]
-    from heudiconv.external.dcmstack import ds
-    mw = ds.wrapper_from_data(dcm.read_file(dcm_fn,
-                                            stop_before_pixels=True,
-                                            force=True))
+    dcm_data = dcm.read_file(dcm_fn, stop_before_pixels=True, force=True)
     # we need to store filenames and acquisition times
     # parse date and time and get it into isoformat
     try:
-        date = mw.dcm_data.ContentDate
-        time = mw.dcm_data.ContentTime.split('.')[0]
+        date = dcm_data.ContentDate
+        time = dcm_data.ContentTime.split('.')[0]
         td = time + date
         acq_time = datetime.strptime(td, '%H%M%S%Y%m%d').isoformat()
     except AttributeError as exc:
         lgr.warning("Failed to get date/time for the content: %s", str(exc))
         acq_time = None
     # add random string
-    randstr = ''.join(map(chr, sample(k=8, population=range(33, 127))))
+    # But let's make it reproducible by using all UIDs
+    # (might change across versions?)
+    randcontent = u''.join(
+        [getattr(dcm_data, f) for f in sorted(dir(dcm_data))
```

Comment:
if the `getattr` returns `None` this will fail. here is a suggestion, if `None` is a possibility.

```suggestion
        [getattr(dcm_data, f) or '' for f in sorted(dir(dcm_data))
```